nuget/lib/dependabot/nuget/update_checker/repository_finder.rb

```diff
@@ -99,6 +114,17 @@ def base_url_from_v3_metadata(metadata)
             &.fetch("@id")
         end
 
+        def registration_url_from_v3_metadata(metadata)
+          allowed_registration_types = %w(
+            RegistrationsBaseUrl/3.0.0-beta
+            RegistrationsBaseUrl
+          )
```

Comment:
According to [this](https://learn.microsoft.com/en-us/nuget/api/registration-base-url-resource#versioning) I'd say set the search order to:

```
RegistrationsBaseUrl/3.6.0
RegistrationsBaseUrl/3.4.0
RegistrationsBaseUrl/3.0.0-rc
RegistrationsBaseUrl/3.0.0-beta
RegistrationsBaseUrl
```